packages/nest/src/schematics/nestjs-schematics/schema.d.ts

```diff
@@ -0,0 +1,28 @@
+export interface NestSchematicsSchema {
+  type:
+    | 'class'
+    | 'controller'
+    | 'decorator'
+    | 'filter'
+    | 'gateway'
+    | 'guard'
+    | 'interceptor'
+    | 'interface'
+    | 'middleware'
+    | 'module'
+    | 'pipe'
+    | 'provider'
+    | 'resolver'
+    | 'service';
+  name: string;
+  project: string;
+  unitTestRunner: 'jest' | 'none';
+  directory?: string;
+  path?: string;
+  flat: boolean;
+  // Additional Nest schematic properties
+  module?: string;
+  skipImport?: string;
+  language?: string;
+  spec?: boolean;
```

Comment:
Might want to add `--dry-run`, do we want to support `--collection` as well so that people can generate using external nest schematics through Nx?

Reply:
What's an example of a nest collection?

Also for an FYI, the `additionalProperties` flag in the schema.json allows for properties to be used that aren't necessarily declared in the schema. SO, dry-run and collection should still work and be passed correctly to the nest schematic programatically.

Reply:
I don't have an example, just referencing the nest docs as that is what Kamil specified under the `--collection` flag.